src/index.ts

```diff
@@ -21,44 +25,85 @@ async function run(): Promise<void> {
+    core.info(`The head commit is: ${pushPayload.head_commit}`);
+    populatePushEventCommitDetails(pushPayload, context);
+  } else if (github.context.eventName === 'pull_request') {
+    const pullRequestPayload = event as PullRequest;
+    populatePullRequestEventCommitDetails(pullRequestPayload, context);
+  }
 
   // TODO add expected description (rather complex to dig up), made by `git describe`
   // Expected description format - "2019-07-18.1563481767-1-g7f60" < 'g<first 4 sha char>'
   //       commit date branched from ^     same sec ^     ^ commit number on branch (0 is the commit you branched from)
   setVariable(EnvironmentVariables.CI_COMMIT_DESCRIPTION, '');
 
   const nowMilliseconds = Date.now();
-  const nowSeconds = Math.floor(nowMilliseconds/1000);
+  const nowSeconds = Math.floor(nowMilliseconds / 1000);
   const nowAsTimeIsoString = new Date(nowMilliseconds).toISOString();
-  const nowAsTimeExpectedIsoString = nowAsTimeIsoString.replace(/\.\d{3}Z/,'Z'); // No millis expect
+  const nowAsTimeExpectedIsoString = nowAsTimeIsoString.replace(
+    /\.\d{3}Z/,
+    'Z',
+  ); // No millis expect
   setVariable(EnvironmentVariables.CI_TIMESTAMP, nowSeconds.toString());
   setVariable(EnvironmentVariables.CI_STRING_TIME, nowAsTimeExpectedIsoString);
 
-
   const branchName = context?.ref?.match(/[^/]+$/)?.[0];
-  if(branchName) {
+  if (branchName) {
     setVariable(EnvironmentVariables.CI_BRANCH, branchName);
   }
 
-  // @ts-ignore - missing lib defs
-  setVariable(EnvironmentVariables.CI_REPO_NAME,  event?.repository?.name );
+  setVariable(EnvironmentVariables.CI_REPO_NAME, event?.repository?.name);
+}
 
-  // TODO The pull request payload is missing in this implementation (tested with PR builds)
-  // Note: the PR functionality in Codeship is not function from my empirical testing, so this behavior is the same as Codeship
-  const pullRequest = event?.pull_request;
-  setVariable(EnvironmentVariables.CI_PULL_REQUEST,  pullRequest?.number?.toString() ?? '');
-  setVariable(EnvironmentVariables.CI_PR_NUMBER,  pullRequest?.html_url ?? 'false');
+function populatePushEventCommitDetails(
+  pushEvent: PushEvent,
+  context: Context,
+) {
+  const headCommit = pushEvent.head_commit;
+  const author = headCommit?.author;
+  setVariable(EnvironmentVariables.CI_COMMIT_ID, context?.sha);
+  setVariable(EnvironmentVariables.CI_COMMIT_MESSAGE, headCommit?.message);
+  setVariable(EnvironmentVariables.CI_COMMITTER_USERNAME, author?.username);
+  setVariable(EnvironmentVariables.CI_COMMITTER_EMAIL, author?.email ?? '');
+  setVariable(EnvironmentVariables.CI_COMMITTER_NAME, author?.name);
+}
+
+async function populatePullRequestEventCommitDetails(
+  pullRequestEvent: PullRequest,
+  context: Context,
+) {
+  const head = pullRequestEvent.head;
+  const user = head.user;
+  setVariable(EnvironmentVariables.CI_COMMIT_ID, head.sha);
+  setVariable(EnvironmentVariables.CI_COMMITTER_USERNAME, user.login);
```

Comment:
Is there a diff b/t `username` for `push` and `login` here for the PR?

Reply:
Yes, unfortunately the models are different for the various event types.  

PushEvent.head_commit gives us a nullable Commit object which has an `author: Committer` object (there is also a `PushEvent.pusher: Committer` that I'm guessing has the same Committer object). The committer object then has  
```
/**
 * For pushes where `after` is or points to a commit object, an expanded representation of that commit. For pushes where `after` refers to an annotated tag object, an expanded representation of the commit pointed to by the annotated tag.
 */
export interface PushEvent {
  ...
  head_commit: Commit | null;
  ...
}

export interface Commit {
  ...
  author: Committer
  ...
}


export interface Committer {
  /**
   * The git author's name.
   */
  name: string;
  /**
   * The git author's email address.
   */
  email: string | null;
  date?: string;
  username?: string;
}
```

PullRequest does not have any Commit object except for an array of them to list all commits on the PR and a User object with the triggering user for the latest commit.  The `PullRequest.head` field provides an object with _some_ of the commit info like the sha and the user information but not the message. 
```
export interface PullRequest {
  ...
  user: User;
  head: {
    label: string;
    ref: string;
    sha: string;
    user: User;
    repo: Repository | null;
  };
  ... 
}

export interface User {
  login: string;
  id: number;
  node_id: string;
  name?: string;
  email?: string | null;
  avatar_url: string;
  gravatar_id: string;
  url: string;
  html_url: string;
  followers_url: string;
  following_url: string;
  gists_url: string;
  starred_url: string;
  subscriptions_url: string;
  organizations_url: string;
  repos_url: string;
  events_url: string;
  received_events_url: string;
  type: "Bot" | "User" | "Organization";
  site_admin: boolean;
}
```